src/backends/x11/meta-output-xrandr.c

```diff
@@ -52,6 +52,32 @@ typedef struct
   int underscan_vborder;
 } MetaOutputXrandr;
 
+static gboolean
+output_is_hdtv (MetaOutput *output)
+{
+  MetaCrtc *crtc;
+  guint width, height;
+
+  crtc = meta_output_get_assigned_crtc (output);
+
+  if (!crtc || !crtc->current_mode)
+    return FALSE;
+
+  /* We must have an HDTV is underscan is already enabled */
+  if (output->is_underscanning)
+    return TRUE;
+
+  if (!g_str_has_prefix (meta_output_get_connector_type_name (output), "HDMI"))
```

Comment:
A bit funny to do it this way rather than `output->connector == META_CONNECTOR_TYPE_HDMIA || output->connector == META_CONNECTOR_TYPE_HDMIB` – doing that in future might make the patch smaller.

Reply:
Indeed.